browser/brave_wallet/solana_provider_browsertest.cc

```diff
@@ -945,10 +945,10 @@ IN_PROC_BROWSER_TEST_F(SolanaProviderTest, ConnectedStatusInIframes) {
   EXPECT_FALSE(IsSolanaConnected(ChildFrameAt(main_frame, 0)));
   EXPECT_TRUE(IsSolanaConnected(ChildFrameAt(main_frame, 1)));
 
-  // navigate second iframe awau will clear its connected status
+  // navigate second iframe away will clear its connected status
   EXPECT_TRUE(
       NavigateIframeToURL(web_contents(), "test-iframe-1", new_iframe_url));
-  EXPECT_TRUE(IsSolanaConnected(ChildFrameAt(main_frame, 1)));
+  EXPECT_FALSE(IsSolanaConnected(ChildFrameAt(main_frame, 1)));
```

Comment:
@darkdh with the `128.0.6613.18` bump it went back to returning `true`. I can revert this change, but I am not sure why EXPECT_TRUE would be the correct expectation here (and contrary to the comment).

Reply:
I noticed that for `test-iframe-0` we explicitly call `CallSolanaDisconnect` and then verify that it's disconnected. Then we navigate `test-iframe-0` to the same page that it already has and check that it's still disconnected.
In the case of `test-iframe-1`, though, we don't call `CallSolanaDisconnect` and just navigate the frame to the same page. I am not sure why in this case the comment says it should be disconnected. If we expect the `BraveWalletProviderDelegateImpl::RenderFrameHostChanged` to trigger, then in this case it doesn't appear to.

Reply:
let's disable that small block
```
diff --git a/browser/brave_wallet/solana_provider_browsertest.cc b/browser/brave_wallet/solana_provider_browsertest.cc
index 4045cb6d3fb..8a97a887caa 100644
--- a/browser/brave_wallet/solana_provider_browsertest.cc
+++ b/browser/brave_wallet/solana_provider_browsertest.cc
@@ -937,6 +937,7 @@ IN_PROC_BROWSER_TEST_F(SolanaProviderTest, ConnectedStatusInIframes) {
   EXPECT_FALSE(IsSolanaConnected(ChildFrameAt(main_frame, 0)));
   EXPECT_TRUE(IsSolanaConnected(ChildFrameAt(main_frame, 1)));

+#if 0
   GURL new_iframe_url =
       https_server_for_files()->GetURL("a.test", "/solana_provider.html");
   // navigate first iframe away won't affect second iframe's connected status
@@ -949,6 +950,7 @@ IN_PROC_BROWSER_TEST_F(SolanaProviderTest, ConnectedStatusInIframes) {
   EXPECT_TRUE(
       NavigateIframeToURL(web_contents(), "test-iframe-1", new_iframe_url));
   EXPECT_TRUE(IsSolanaConnected(ChildFrameAt(main_frame, 1)));
+#endif
 }
```
The current checking for the first iframe is wrong because as you mentioned, it is disconnected explicitly before.
Apparently, iframe cases don't have the same behavior like main frame navigate away in `SolanaProviderTest.ConnectedStatusWithDocumentChanged`

Reply:
Sounds good. Done in https://github.com/brave/brave-core/pull/24227/commits/683d82543978426a4b8c3044aac83c095a75d0f3